fuel/app/classes/controller/widgets.php

```diff
@@ -43,22 +43,34 @@ public function get_all()
 	 */
 	public function get_detail()
 	{
-		$widget = DB::select()
-			->from('widget')
-			->where('id', $this->param('id'))
-			->execute();
+		$widget = new Materia\Widget();
+		$loaded = $widget->get($this->param('id'));
+
+		if ( ! $loaded) throw new HttpNotFoundException;
 
-		if ( ! $widget) throw new HttpNotFoundException;
+		$demo = $widget->meta_data['demo'];
+		$play_id = \Materia\Api::session_play_create($demo, false);
```

Comment:
this isn't ideal since it automatically assumes you're going to play the demo. would be better to do this once the "play demo" button is pressed